packages/hash/frontend/src/components/GlideGlid/utils/interactable-manager.ts

```diff
@@ -0,0 +1,161 @@
+   * ```ts
+   * const interactableStore = {
+   *  "users-1-2": { button1: { ...interactable } },
+   * };
+   * ```
+   */
+  private interactableStore: Record<CellPath, Record<string, Interactable>> =
+    {};
+
+  /**
+   * @param args Draw args of cell
+   * @param props Properties which will will be used to create the `Interactable`
+   * @returns The created `Interactable`
+   */
+  create(
+    args: DrawArgs<CustomCell>,
+    props: Omit<Interactable, "hovered" | "path" | "cellRect">,
+  ): Interactable {
+    // used -100 to prevent handling events on non-hovered cells
+    const { hoverX = -100, hoverY = -100, rect } = args;
+
+    const hovered = isCursorOnInteractable(
+      { posX: hoverX, posY: hoverY },
+      props.pos,
+      rect,
+    );
+
+    return {
+      ...props,
+      hovered,
+      cellRect: rect,
+      path: drawArgsToPath(args),
+    };
+  }
+
+  /**
+   * Call this function at the end of the `draw` function of the custom cell to register the interactables.
+   * @param args Draw args of cell
+   * @param interactables List of the interactables for a specific cell.
+   */
+  setInteractablesForCell(
+    args: DrawArgs<CustomCell>,
+    interactables: Interactable[],
+  ) {
+    const path = drawArgsToPath(args);
+
+    /**
+     * for each interactable, check if the hover status changed
+     * if it's changed, trigger corresponding event
+     */
+    for (const interactable of interactables) {
+      const existing = this.interactableStore[path]?.[interactable.id];
+
+      if (existing && existing.hovered !== interactable.hovered) {
+        const event = interactable.hovered
+          ? interactable.onMouseEnter
+          : interactable.onMouseLeave;
+
+        event?.(interactable);
+      }
+    }
+
+    const interactableMap = Object.fromEntries(
+      interactables.map((interactable) => [interactable.id, interactable]),
+    );
+
+    this.interactableStore[path] = interactableMap;
+  }
+
+  /**
+   * Checks if the cursor position is overlapping with an `Interactable` inside clicked cell.
+   * If it's overlapping, triggers the corresponding event handler for the clicked `Interactable`
+   * @returns true if handled the click event, false if not
+   */
+  handleClick(path: CellPath, event: CursorPos): boolean {
+    const interactableMap = this.interactableStore[path] ?? {};
+    const interactables = Object.values(interactableMap);
+
+    let foundInteractable: Interactable | undefined;
+
+    for (let i = 0; i < interactables.length; i++) {
+      const interactable = interactables[i]!;
+      const { pos, cellRect } = interactable;
+
+      const hovered = isCursorOnInteractable(event, pos, cellRect);
+
+      if (hovered) {
+        foundInteractable = interactable;
+      }
+    }
+
+    if (!foundInteractable) {
+      return false;
+    }
+
+    const { onClick, onMouseEnter } = foundInteractable;
+
+    const handler = onClick ?? onMouseEnter;
```

Comment:
Why would we want to call `onMouseEnter` on a click handler?

Reply:
Comment added in https://github.com/hashintel/hash/pull/1386/files/70dfc7c109e1d2ab3b2045315fd8e04250edd1dd..abb7cfc76043b432b5e3e123b554db8376eafbda